pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -742,6 +742,19 @@ func findNextInitContainerToRun(pod *v1.Pod, podStatus *kubecontainer.PodStatus)
 		return nil, nil, true
 	}
 
+	// If any of the main containers have status, then all init containers must
+	// after executed at some point in the past.  However, they could be removed
+	// from the container runtime now, and if we proceed, it would appear as if they
+	// never ran and will re-execute improperly.
+	for i := range pod.Spec.Containers {
```

Comment:
should be InitContainers on line 749 and line 750, right?

Reply:
no, we are wondering if there is status for an _main_ containers